src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,258 @@
+            false
+        );
+        assets[TUSD] = Asset(
+            ERC20(TUSD).balanceOf(AaveV2Ethereum.COLLECTOR),
+            75,
+            0xec746eCF986E2927Abd291a2A1716c940100f8Ba,
+            ERC20(TUSD).decimals(),
+            8,
+            false
+        );
+        assets[ATUSD] = Asset(
+            ERC20(ATUSD).balanceOf(AaveV2Ethereum.COLLECTOR),
+            75,
+            0xec746eCF986E2927Abd291a2A1716c940100f8Ba,
+            ERC20(ATUSD).decimals(),
+            8,
+            false
+        );
+        assets[AMANA] = Asset(
+            ERC20(AMANA).balanceOf(AaveV2Ethereum.COLLECTOR),
+            200,
+            0x56a4857acbcfe3a66965c251628B1c9f1c408C19,
+            ERC20(AMANA).decimals(),
+            8,
+            false
+        );
+        assets[MANA] = Asset(
+            ERC20(MANA).balanceOf(AaveV2Ethereum.COLLECTOR),
+            200,
+            0x56a4857acbcfe3a66965c251628B1c9f1c408C19,
+            ERC20(MANA).decimals(),
+            8,
+            false
+        );
+        assets[ABUSD] = Asset(
+            ERC20(ABUSD).balanceOf(AaveV2Ethereum.COLLECTOR),
+            75,
+            0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A,
+            ERC20(ABUSD).decimals(),
+            8,
+            false
+        );
+        assets[BUSD] = Asset(
+            ERC20(BUSD).balanceOf(AaveV2Ethereum.COLLECTOR),
+            75,
+            0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A,
+            ERC20(BUSD).decimals(),
+            8,
+            false
+        );
+        assets[ZRX] = Asset(
+            ERC20(ZRX).balanceOf(AaveV2Ethereum.COLLECTOR),
+            300,
+            0x2885d15b8Af22648b98B122b22FDF4D2a56c6023,
+            ERC20(ZRX).decimals(),
+            8,
+            false
+        );
+        assets[AZRX] = Asset(
+            ERC20(AZRX).balanceOf(AaveV2Ethereum.COLLECTOR),
+            300,
+            0x2885d15b8Af22648b98B122b22FDF4D2a56c6023,
+            ERC20(AZRX).decimals(),
+            8,
+            false
+        );
+        assets[AENS] = Asset(
+            ERC20(AENS).balanceOf(AaveV2Ethereum.COLLECTOR),
+            300,
+            0x5C00128d4d1c2F4f652C267d7bcdD7aC99C16E16,
+            ERC20(AENS).decimals(),
+            8,
+            false
+        );
+        assets[AUST] = Asset(
+            ERC20(AUST).balanceOf(AaveV2Ethereum.COLLECTOR),
+            200,
+            0xa20623070413d42a5C01Db2c8111640DD7A5A03a,
+            ERC20(AUST).decimals(),
+            18,
+            false
+        );
+
+        USDC_DECIMALS = USDC.decimals();
+        ETH_USD_ORACLE_DECIMALS = AggregatorV3Interface(ETH_USD_FEED).decimals();
+    }
+
+    /// @notice Swaps USDC for specified token
+    /// @param _token the address of the token to swap USDC for
+    /// @param _amountOut the amount of token wanted
+    /// @dev User has to approve USDC transfer prior to calling swap
+    function swap(address _token, uint256 _amountOut) external {
```

Comment:
Small nitpick. And no major opinion here. I think this function should be called `purchase` instead of `swap`. It's more reflective of what's actually happening in this function logic.